src/hera/workflows/script.py

```diff
@@ -591,11 +575,8 @@ def _extract_all_output_annotations(source: Callable) -> List:
     output = []
 
     for _, func_param in inspect.signature(source).parameters.items():
-        if get_origin(func_param.annotation) is Annotated:
-            annotation_args = get_args(func_param.annotation)
-            annotated_type = annotation_args[1]
-            if isinstance(annotated_type, (Artifact, Parameter)) and annotated_type.output:
-                output.append(annotation_args)
+        if (annotated := get_annotated_metadata(func_param.annotation, (Artifact, Parameter))) and annotated.output:
+            output.append(annotated)
```

Comment:
Agree. great catch. Updated to append annotation_args again.
For the `_extract_all_output_annotations`, it is used when building Script template, so we cannot delete it.

Reply:
Ha, my mistake! I did a search and it only showed the definition; I missed the small "and one more"

Reply:
This was a small bug fix I asked for in https://github.com/argoproj-labs/hera/pull/1168/files#r1734309545

We don't use the list (for now) but it would make more sense to get the `Artifact`/`Parameter` metadata rather than ALL the metadata (my bad from previous implementation 😆)